### PR TITLE
Updates for DDN Alpha and TypeScript Deno connector compatibility

### PR DESCRIPTION
**Pull request by daniel-chambers**
This PR updates this repo to be compatible with the DDN Alpha and also with the latest TypeScript Deno connector compatibility (namely the improved watch support, introduced in https://github.com/hasura/ndc-typescript-deno/pull/79).

* Added a workaround that ensures the `errors` property is not nullable/undefined on the return object, because the TypeScript Deno connector doesn't support nullable/undefined types properly yet
* Dockerfile updated to match latest TypeScript Deno version
* `config.json` file deleted and an empty config just inlined into the CLI commands in the readme
* The metadata.hml has been replaced with a full DDN project structure (hasura.yaml, build-profile.yaml, etc)
* Readme has been updated with latest Hasura CLI syntax
* VSCode `extensions.json` has been added to recommend the required VSCode extensions
* VSCode `settings.json` added to ensure Deno extension is activated in this repo

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "deno.enable": true,
+  "deno.lint": true,
+  "deno.unstable": false
```
Comment (sordina) [inline .vscode/settings.json:4]: We're using unstable features in the connector to support the NPM modules so should we try to match this with the dev tools?

File: Dockerfile [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
 # It is recommended that you pin a release tag rather than using `main`
 
 FROM ghcr.io/hasura/ndc-typescript-deno:main
 
-COPY ./functions:/functions
-RUN EARLY_ENTRYPOINT_EXIT=true ./entrypoint.sh /app/ndc-typescript-deno --configuration /placeholder-config.json
+COPY ./functions /functions/src
+
+# Pre-cache inference results and dependencies
+RUN PRECACHE_ONLY=true /app/entrypoint.sh
```
Comment (sordina) [inline Dockerfile:9]: 👍

**Review by sordina — APPROVED** [review]
Looks good to me. The subgraph stuff certainly adds a lot of files, but given that this is primarily an example repo I think that's fine. If only there was some way to make sure it stayed in sync with any future format changes...

File: .vscode/settings.json (hunk 1 above)
Comment (daniel-chambers) [reply .vscode/settings.json:4]: I'm not sure that refers to that, I think this enables access to Deno's unstable APIs. I've used an NPM package (for testing) in this project with that set to false and it works just fine.

**Comment by daniel-chambers** [comment]
I tested this against DDN Production and it returns a response that is making the engine choke. 

image

More investigation required before merging.

**Pull request merged**